docs/source/release_notes/release-notes.rst

```diff
@@ -63,6 +85,8 @@ develop
            Previously, clients would immediately retry the connection on the node with a 503 two times (for a total of three attempts) before failing over.
```

Comment:
I'll rephrase "AtlasDB HTTP clients" to "Proxies created via `AtlasDbHttpClients`", and also explicitly say Timelock clients in the second line